src/contradictory_claims/data/preprocess_cord.py

```diff
@@ -27,7 +34,147 @@ def filter_metadata_for_covid19(metadata_path: str, virus_lex_path: str, pub_dat
+                                               | (filename == covid19_metadata.pmc_json_files))  # noqa: W503
+                            cord_uid = list(covid19_metadata.loc[check_file_name, 'cord_uid'])[0]
+                            try:
+                                text = temp_json['abstract'][0][0]['text']
+                                section = temp_json['abstract'][0][0]['section']
+                                for key, v in replace_dict.items():
+                                    text = text.replace(key, v)
+                                    section = section.replace(key, v)
+                                covid19_dict[k] = {'cord_uid': cord_uid,
+                                                   'sentence': text,
+                                                   'section': section}
+                                k = k + 1
+                            except KeyError:
+                                pass
+
+                            for temp_dict in temp_json['body_text'][0]:
+                                text = temp_dict['text']
+                                section = temp_dict['section']
+                                for key, v in replace_dict.items():
+                                    text = text.replace(key, v)
+                                    section = section.replace(key, v)
+                                covid19_dict[k] = {'cord_uid': cord_uid,
+                                                   'sentence': text,
+                                                   'section': section}
+                                k = k + 1
+
+#        if iter_num%100==0:
+#            print('Number of files read:', iter_num)
+
+    return pd.DataFrame.from_dict(covid19_dict, orient='index')
+
+
+def construct_regex_match_pattern(search_terms: List[str], search_type: bool = 0):
+    """
+    Construct regex search pattern for the specified terms.
+
+    :param terms_dict: list of search terms
+    :param search_type: 1 = exact pattern, 0 = fuzzy pattern
+    :return: Regex search pattern
+    """
+    if search_type == 1:
+        exact_pattern = '|'.join(search_terms)
+
+        return exact_pattern
+
+    else:
+        # TODO: fix flake8 error code FS001
+        fuzzy_terms = ['.*%s.*' % i for i in search_terms]  # noqa: FS001
+        fuzzy_pattern = '|'.join(fuzzy_terms)
+
+        return fuzzy_pattern
+
+
+def extract_regex_pattern(section_list: List[str], pattern: str):
+    """
+    Extract list of section names that match the specified regex pattern.
+
+    :param section_list: list of section names to search in
+    :param pattern: regex pattern to search for
+    :return: List of extracted section names
+    """
+    r = re.compile(pattern, re.IGNORECASE)
+    extracted_list = list(filter(r.match, section_list))
+    # remaining_list = list(set(section_list) - set(extracted_list))
+
+    return extracted_list
```

Comment:
Sorry this is slightly unclear to me.  The user inputs a list of section names that appear in a paper (e.g. [Introduction, Results, Discussion]) and a regex (e.g. "Result") to retrieve the section headers they want?

Reply:
Yeah. The input is a list of all section headers and the regex search pattern to look for in the section header. Output is the subset of section names that matches the regex pattern.